### DISCUSSION TITLE
How to send notification without a sound

### DISCUSSION THREAD

**Discussion by mario-van-zadel** [kreait/firebase-php#587] [Q&A]
In my android app, the user has the possibility to disable notification sounds in the app settings.

If `$withSound` is set, the method `withDefaultSounds` is called.

```php
$data = [
    'type' => 'chat-message',
    'notification_foreground' => true
];

$message = Messaging\CloudMessage::withTarget(Messaging\MessageTarget::TOKEN, $pushToken)
    ->withNotification(Messaging\Notification::create($title, $body))
    ->withData($data);

if ($withSound) {
    $message = $message->withDefaultSounds();
}

$messaging->send($message);
```

But a sound is always played when receiving a notification.
Also explicit setting `sound` to `false` doesn't work.

```php
if ($withSound)) {
    $message = $message->withDefaultSounds();
} else {
    $message = $message
        ->withAndroidConfig(
            Messaging\AndroidConfig::new()
                ->withSound('false')
        );
}
```

I debugged it to make sure that `withDefaultSounds` isn't called.

Does anyone have an idea, what I'm doing wrong?
I thought that not calling the method `withDefaultSounds` should disable notification sounds.

**Comment by jeromegamez** [comment]
That's a good question! 

First and foremost: I don't know for sure 😅.

As far as I know, whether a sound is actually played on the device or not, is controlled by the device, not by the message. The message only specified: "If a sound is played, use this one", then, on the device, the system looks at the notification settings and if it's in DND mode and then acts accordingly. If the message has no sound specified, the device then uses the system default sound.

If you want to make sure that no sound is played, I think the solution could be to send a background message instead of a notification message by not using the `notification` methods and fields, and instead put something like `notification_title`and `notification_body` into the `data`array.

Then, when the message reaches your app, the app handles the message and triggers/creates the notification on the device. 🤞

**Reply by mario-van-zadel** [reply]
Thank you for the great explanation, @jeromegamez!

> As far as I know, whether a sound is actually played on the device or not, is controlled by the device, not by the message. The message only specified: "If a sound is played, use this one", then, on the device, the system looks at the notification settings and if it's in DND mode and then acts accordingly. If the message has no sound specified, the device then uses the system default sound.

I've tested it with another device. It seems that you're totally right. It's device-dependent and sending a push notification without a sound was working on that device.

> If you want to make sure that no sound is played, I think the solution could be to send a background message instead of a notification message by not using the notification methods and fields, and instead put something like notification_titleand notification_body into the dataarray.

> Then, when the message reaches your app, the app handles the message and triggers/creates the notification on the device. 🤞

Thank you for suggesting this alternative. It seems to be a great alternative to create the "visible" notifications on the device side. I'll check how I can handle this kind of notifications because it's an Ionic App.
